From b90c05bec6e45cc07630889318faae227b829af6 Mon Sep 17 00:00:00 2001
From: Jonathan Dayton <jdayton@microsoft.com>
Date: Mon, 29 Apr 2024 12:30:44 -0500
Subject: [PATCH] Ignore whitespace differences when testing error messages

---
 test/test_bin.py | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/test/test_bin.py b/test/test_bin.py
index acee10f..9ae752b 100644
--- a/test/test_bin.py
+++ b/test/test_bin.py
@@ -75,20 +75,32 @@ class TestError(unittest.TestCase):
         'invalid JSON in %s: Expecting value: line 1 column 1 (char 0)'
         % BAD_JSON_FILE)
 
+    def assertEqualIgnoreWhitespace(
+            self,
+            first: str,
+            second: str,
+            msg: "str|None" = None,
+        ):
+        return self.assertEqual(
+            " ".join(first.split()),
+            " ".join(second.split()),
+            msg=msg
+        )
+
     def test_no_input(self):
         (stdout, stderr) = run()
-        self.assertEqual(stderr, stderr_message(
+        self.assertEqualIgnoreWhitespace(stderr, stderr_message(
             'noting to do - no schemas or objects given'))
         self.assertEqual(stdout, '')
 
     def test_object_not_json(self):
         (stdout, stderr) = run([self.BAD_JSON_FILE])
-        self.assertEqual(stderr, self.BAD_JSON_MESSAGE)
+        self.assertEqualIgnoreWhitespace(stderr, self.BAD_JSON_MESSAGE)
         self.assertEqual(stdout, '')
 
     def test_schema_not_json(self):
         (stdout, stderr) = run(['-s', self.BAD_JSON_FILE])
-        self.assertEqual(stderr, self.BAD_JSON_MESSAGE)
+        self.assertEqualIgnoreWhitespace(stderr, self.BAD_JSON_MESSAGE)
         self.assertEqual(stdout, '')
 
 